app/src/main/java/io/heckel/ntfy/msg/DownloadAttachmentWorker.kt

```diff
@@ -102,8 +106,8 @@ class DownloadWorker(private val context: Context, params: WorkerParameters) : W
                             save(attachment.copy(progress = progress))
                             lastProgress = System.currentTimeMillis()
                         }
-                        if (contentLength != null && bytesCopied > contentLength) {
-                            throw Exception("Attachment is longer than response headers said.")
+                        if (downloadLimit != null && bytesCopied > downloadLimit) {
+                            throw Exception("Attachment is longer than max download size.")
```

Comment:
Looking at the [original code](https://github.com/binwiederhier/ntfy-android/blob/main/app/src/main/java/io/heckel/ntfy/msg/DownloadWorker.kt), I didn't change anything regarding the `userAction`

But I agree that the `userAction` override should maybe set the `downloadLimit` to `null`

Reply:
You didnt change the original code, but the original would only bail out if the content length that was returned didnt match the actual bytes. Now you do that for the download limit. if i understand correctly, thats a behavioral change. Baby brain may affect me though.

Reply:
Looking back through [Discord](https://discordapp.com/channels/874398661709295626/874398661709295629/997676553053945977), we agreed to change the logic since the `Content-Length` header isn't always passed. (It's not just your baby brain because I also forgot about that conversation, haha)

You're right, though, that with the way the code is right now, a user can only force the download if it's also less than the max download size, so it's not a true "force download" behavior. I'll change the code to remove the download limit if `userAction == true`